torchgeo/models/convlstm.py

```diff
@@ -0,0 +1,230 @@
+
+        Args:
+            input_dim: Number of channels in the input.
+            hidden_dim: Number of hidden channels. Can be a single int (for all
+                layers) or a sequence of ints (one for each layer).
+            kernel_size: Size of the convolutional kernel. Can be a single tuple
+                (for all layers) or a sequence of tuples (one for each layer).
+            num_layers: Number of LSTM layers stacked on each other.
+            batch_first: If ``True``, then the input and output tensors are
+                provided as (b, t, c, h, w).
+            bias: If ``True``, adds a learnable bias to the output.
+            return_all_layers: If ``True``, will return the list of computations
+                for all layers.
+        """
+        super().__init__()
+
+        # Normalize hidden_dim to a list of ints
+        if isinstance(hidden_dim, int):
+            self.hidden_dim: list[int] = [hidden_dim] * num_layers
+        else:
+            self.hidden_dim = list(hidden_dim)
+
+        # Normalize kernel_size to a list of tuples
+        if isinstance(kernel_size, tuple):
+            self.kernel_size: list[tuple[int, int]] = [kernel_size] * num_layers
+        elif isinstance(kernel_size, list):
+            self.kernel_size = [
+                ks if isinstance(ks, tuple) else (ks, ks) for ks in kernel_size
+            ]
+        else:
+            raise ValueError(
+                '`kernel_size` must be an int, a tuple, or a list of ints/tuples.'
+            )
+
+        if not len(self.kernel_size) == len(self.hidden_dim) == num_layers:
+            raise ValueError('Inconsistent list length.')
+
+        self.input_dim = input_dim
+        self.num_layers = num_layers
+        self.batch_first = batch_first
+        self.bias = bias
+        self.return_all_layers = return_all_layers
+
+        cell_list = []
+        for i in range(self.num_layers):
+            cur_input_dim = self.input_dim if i == 0 else self.hidden_dim[i - 1]
+            cell_list.append(
+                ConvLSTMCell(
+                    input_dim=cur_input_dim,
+                    hidden_dim=self.hidden_dim[i],
+                    kernel_size=self.kernel_size[i],
+                    bias=self.bias,
+                )
+            )
+
+        self.cell_list = nn.ModuleList(cell_list)
+
+    def forward(
+        self,
+        input_tensor: torch.Tensor,
+        hidden_state: list[tuple[torch.Tensor, torch.Tensor]] | None = None,
+    ) -> tuple[list[torch.Tensor], list[tuple[torch.Tensor, torch.Tensor]]]:
+        """Forward pass of the ConvLSTM.
+
+        Args:
+            input_tensor: A 5-D Tensor of shape (t, b, c, h, w) or (b, t, c, h, w).
+            hidden_state: An optional initial hidden state.
+
+        Returns:
+            A tuple of two lists:
+            1. layer_output_list: List of Tensors of shape (b, t, c, h, w).
+            2. last_state_list: List of tuples of (h, c) for the last time step.
```

Comment:
Sphinx does not like this, maybe add a blank line before the start of the numbered list, play around with indentation, or just simplify the formatting a bit.

Reply:
Looks good now: https://torchgeo--2898.org.readthedocs.build/en/2898/api/models.html#torchgeo.models.ConvLSTM.forward